parquet-variant-compute/src/variant_array.rs

```diff
@@ -149,6 +150,154 @@ impl VariantArray {
+                    current_variant = current_variant.get_list_element(*idx)?;
+                }
+            }
+        }
+
+        Some(current_variant)
+    }
+
+    /// Get values at multiple paths for the variant at the given index
+    ///
+    /// Convenience method that applies `get_path()` to multiple paths at once.
+    /// Useful for extracting multiple fields from a single variant row.
+    pub fn get_paths(
+        &self,
+        index: usize,
+        paths: &[crate::field_operations::VariantPath],
+    ) -> Vec<Option<parquet_variant::Variant>> {
+        let mut results = Vec::new();
+        for path in paths {
+            results.push(self.get_path(index, path));
+        }
+        results
+    }
+
+    /// Get the field names for an object at the given index
+    pub fn get_field_names(&self, index: usize) -> Vec<String> {
+        if index >= self.len() {
+            return vec![];
+        }
+
+        if self.is_null(index) {
+            return vec![];
+        }
+
+        let variant = self.value(index);
+        if let Some(obj) = variant.as_object() {
+            let mut paths = Vec::new();
+            for i in 0..obj.len() {
+                if let Some(field_name) = obj.field_name(i) {
+                    paths.push(field_name.to_string());
+                }
+            }
+            paths
+        } else {
+            vec![]
+        }
+    }
+
+    /// Extract field values by path from all variants in the array
+    ///
+    /// Applies `get_path()` to a single path across all rows in the array.
+    /// Useful for extracting a column of values from nested variant data.
+    pub fn extract_field_by_path(
+        &self,
+        path: &crate::field_operations::VariantPath,
+    ) -> Vec<Option<parquet_variant::Variant>> {
+        let mut results = Vec::new();
+        for i in 0..self.len() {
+            results.push(self.get_path(i, path));
+        }
+        results
+    }
+
+    /// Create a new VariantArray with a field removed from all variants
+    pub fn with_field_removed(&self, field_name: &str) -> Result<Self, ArrowError> {
+        let mut builder = crate::variant_array_builder::VariantArrayBuilder::new(self.len());
+
+        for i in 0..self.len() {
+            if self.is_null(i) {
+                builder.append_null();
+            } else {
+                match FieldOperations::remove_field_bytes(
+                    self.metadata(i),
+                    self.value_bytes(i),
+                    field_name,
+                )? {
+                    Some(new_value) => {
+                        builder.append_variant_buffers(self.metadata(i), &new_value);
+                    }
+                    None => {
+                        // Field didn't exist, use original value
+                        builder.append_variant_buffers(self.metadata(i), self.value_bytes(i));
+                    }
+                }
```

Comment:
I think there's some redundancy here?
```suggestion
                let new_value = FieldOperations::remove_field_bytes(
                    self.metadata(i),
                    self.value_bytes(i),
                    field_name,
                )?;
                // Use original value if the field didn't exist
                let new_value = new_value.as_ref().unwrap_or_else(|| self.value_bytes(i));
                builder.append_variant_buffers(self.metadata(i), new_value);
```
(again below)